File: Competetive programming/Python/Unit 4/Shoemakers_Problem.py
print("===== SHOEMAKER'S PROBLEM =====\n")

n = int(input("Enter number of jobs: "))

jobs = []

for i in range(n):

    print(f"\nJob {i+1}")

    time = int(input("Time Required: "))
    fine = int(input("Daily Fine: "))

    jobs.append((i+1, time, fine))

jobs.sort(key=lambda x: (-x[2]/x[1], x[0]))

print("\nOptimal Job Order:")

for job in jobs:
    print(job[0], end=" ")

print()

File: Competetive programming/C++/Unit 4/Shoemakers_Problem.cpp
#include <iostream>
using namespace std;

struct Job {
    int id;
    int time;
    int fine;
};

int main() {

    int n;

    cout << "===== SHOEMAKER'S PROBLEM =====\n\n";

    cout << "Enter number of jobs: ";
    cin >> n;

    Job jobs[100];

    cout << "\nEnter Time and Fine for each job.\n";
    cout << "Example: Time=3 Fine=4\n\n";

    for(int i = 0; i < n; i++) {

        jobs[i].id = i + 1;

        cout << "Job " << i + 1 << " Time: ";
        cin >> jobs[i].time;

        cout << "Job " << i + 1 << " Fine: ";
        cin >> jobs[i].fine;
    }

    for(int i = 0; i < n - 1; i++) {

        for(int j = i + 1; j < n; j++) {

            if(jobs[i].fine * jobs[j].time <
               jobs[j].fine * jobs[i].time) {

                Job temp = jobs[i];
                jobs[i] = jobs[j];
                jobs[j] = temp;
            }
        }
    }

    cout << "\nOptimal Job Order:\n";

    for(int i = 0; i < n; i++) {
        cout << jobs[i].id << " ";
    }

    cout << endl;

    return 0;
}